src/main/java/de/codingair/clansystem/spigot/base/managers/ClanManager.java

```diff
@@ -0,0 +1,171 @@
+            HandlerList.unregisterAll(this.listener);
+            this.listener = null;
+        }
+    }
+
+    @Override
+    public void save(boolean autoSaver) {
+        //todo: save all to db
+    }
+
+    private void save(Clan clan) {
+        //todo: save clan to db
+    }
+
+    public void onJoin(Player player) {
+        SpigotClan c = getClan(player);
+        if(c != null) this.waiting.remove(prepareName(c.getName(), true));
+        else {
+            //try to load
+            loadClan(player, null);
+        }
+    }
+
+    public void onQuit(Player player) {
+        SpigotClan c = getClan(player);
+
+        if(c != null) {
+            if(!c.isSomebodyOnline(player)) {
+                this.waiting.add(prepareName(c.getName(), true), TIME_OUT);
+            }
+        }
+    }
+
+    private void loadClan(String name, @Nullable Callback<SpigotClan> callback) {
+        SpigotClan c = getClan(name);
+        if(c != null) {
+            if(callback != null) callback.accept(c);
+            return;
+        }
+
+        loading.add(name);
+
+        /*
+            TODO: LOADING
+            - load SpigotClan
+            - add to cache
+            - remove SpigotClan name from loading list
+            - trigger callback IF NOT NULL
+         */
+    }
+
+    private void loadClan(Player player, @Nullable Callback<SpigotClan> callback) {
+        SpigotClan c = getClan(player);
+        if(c != null) {
+            if(callback != null) callback.accept(c);
+            return;
+        }
+
+        loading.add(player);
+
+        /*
+            TODO: LOADING
+            - load SpigotClan
+            - add to cache
+            - remove player from loading list
+            - trigger callback IF NOT NULL
+         */
+    }
+
+    private void loadClan(Collection<? extends Player> players) {
+        players.removeIf(p -> getClan(p) != null);
+        if(players.isEmpty()) return;
+
+        loading.addAll(players);
+
+        /*
+            TODO: LOADING
+            - load clans
+            - add to cache
+            - remove players and SpigotClan names from loading list
+         */
+    }
+
+    public SpigotClan getClan(Player player) {
+        return playerClans.get(player);
+    }
+
+    public SpigotClan getClan(String name) {
+        return clans.get(prepareName(name, true));
+    }
+
+    public boolean exists(String name) {
+        return getClan(name) != null;
+    }
+
+    public static String prepareName(String clanName, boolean asKey) {
+        if(clanName == null) return null;
+
+        clanName = ChatColor.translateAlternateColorCodes('&', clanName);
+        clanName = ChatColor.stripColor(clanName);
+        clanName.trim();
+        if(asKey) clanName = clanName.replace(" ", "_");
+
+        return clanName;
+    }
+
+    public boolean isPending(Player player) {
+        return loading.contains(player);
+    }
+
+    public boolean isPending(String SpigotClan) {
```

Comment:
this is not great considering the String is changeable, using SpigotClan would be better in terms of logic (a String can contain anything), but then we would need a SpigotClan.equals() method.

We should go back here when the database is designed an implemented.